api/src/main/java/edu/wpi/first/shuffleboard/api/sources/recording/Marker.java

```diff
@@ -0,0 +1,94 @@
+package edu.wpi.first.shuffleboard.api.sources.recording;
+
+import java.util.Objects;
+
+/**
+ * Marks an event in a recording.
+ */
+public final class Marker {
+
+  private final String name;
+  private final String description;
+  private final MarkerImportance importance;
+  private final long timestamp;
+
+  /**
+   * Creates a new event marker.
+   *
+   * @param name        the name of the marked event
+   * @param description a description of the marked event
+   * @param importance  the importance of the event
+   * @param timestamp   the time the event occurred, measured in milliseconds since the start of the recording
+   */
+  public Marker(String name, String description, MarkerImportance importance, long timestamp) {
+    this.name = Objects.requireNonNull(name, "name");
+    this.description = Objects.requireNonNull(description, "description");
+    this.importance = Objects.requireNonNull(importance, "importance");
+    this.timestamp = timestamp;
+  }
+
+  /**
+   * Creates a new event marker with no description.
+   *
+   * @param name       the name of the marked event
+   * @param importance the importance of the event
+   * @param timestamp  the time the event occurred, measured in milliseconds since the start of the recording
+   */
+  public Marker(String name, MarkerImportance importance, long timestamp) {
+    this(name, "", importance, timestamp);
+  }
```

Comment:
I'm going to reiterate what I've suggested before, it's always better to not expose constructors directly and instead expose static factory methods.